tests/JavaTest.java

```diff
@@ -506,10 +514,338 @@ static void TestFixedLengthArrays() {
         TestEq(table.a().d(nested, 1).c(1), TestEnum.B);
     }
 
+
+    public static void FlexBuffersTest() {
+        FlexBuffersBuilder builder = new FlexBuffersBuilder(ByteBuffer.allocate(512),
+                FlexBuffersBuilder.BUILDER_FLAG_SHARE_KEYS_AND_STRINGS);
+
+        // Write the equivalent of:
+        // { vec: [ -100, "Fred", 4.0, false ], bar: [ 1, 2, 3 ], bar3: [ 1, 2, 3 ],
+        // foo: 100, bool: true, mymap: { foo: "Fred" } }
+        // It's possible to do this without std::function support as well.
+        builder.putMap(slb -> {
+            slb.putVector("vec", vb -> {
```

Comment:
this lambda based interface is nice, but likely expensive. Maybe instead just do a `startVector` / `endVector` based API?

Reply:
I hope you're not annoyed by all my requests for efficiency, but these data formats are about efficiency first, convenience second :)

Reply:
Yeah the idea was also to make it less error prone with the "helper classes". But maybe we can have a bare-bones high perf API, and another builder with somewhat easier to use / more expensive option.

I am open to suggestions

Reply:
Definitely prefer to keep the base API as efficient as possible. I'd only add higher level APIs once we have examples where the base API is really too cumbersome.